ml-agents/mlagents/trainers/torch/model_serialization.py

```diff
@@ -66,15 +66,18 @@ def __init__(self, policy):
             + [f"visual_observation_{i}" for i in range(self.policy.vis_obs_size)]
             + ["action_masks", "memories"]
         )
+        self.dynamic_axes = {name: {0: "batch"} for name in self.input_names}
 
         self.output_names = ["version_number", "memory_size"]
         if self.policy.behavior_spec.action_spec.continuous_size > 0:
             self.output_names += [
                 "continuous_actions",
                 "continuous_action_output_shape",
             ]
+            self.dynamic_axes.update({"continuous_actions": {0: "batch"}})
         if self.policy.behavior_spec.action_spec.discrete_size > 0:
             self.output_names += ["discrete_actions", "discrete_action_output_shape"]
+            self.dynamic_axes.update({"discrete_actions": {0: "batch"}})
```

Comment:
Since the output will actually be discrete_actions_log_probs instead of discrete_actions, we need to rename this output.

Reply:
I think this will have to keep the same name for backward compat concern(old C# + new python), and we give the actual discrete action a new name like _sampled_discrete_actions_